docs/pipeline-components-and-applications/enrichment-components/4-0-x-upgrade-guide/index.md

```diff
@@ -0,0 +1,62 @@
+---
+title: "4.0.x upgrade guide"
+sidebar_position: 0
+---
+
+## Breaking changes
+
+### New license
+
+Since version 4.0.0, Enrich has been migrated to use the [Snowplow Limited Use License](https://docs.snowplow.io/limited-use-license-1.0/) ([FAQ](/docs/contributing/limited-use-license-faq/index.md)).
+
+## Upgrading
+
+### License acceptance
+
+You have to explicitly accept the [Snowplow Limited Use License](https://docs.snowplow.io/limited-use-license-1.0/) ([FAQ](/docs/contributing/limited-use-license-faq/index.md)). To do so, either set the `ACCEPT_LIMITED_USE_LICENSE=yes` environment variable, or update the following section in the configuration:
+
+```hcl
+{
+  license {
+    accept = true
+  }
+  ...
+}
+```
+
+### Atomic fields limits
+
+Since version 4.0.0 it's possible to define custom atomic (#TODO docs link?) fields (strings) length limits in configuration.
+
+Example:
+```hcl
+{
+  ...
+  # Optional. Configuration section for various validation-oriented settings.
+  "validation": {
+   
+    # Optional. Configuration for custom maximum atomic fields (strings) length.
+    # Map-like structure with keys being field names and values being their max allowed length
+    "atomicFieldsLimits": {
+        "app_id": 5
+        "mkt_clickid": 100000
+        # ...and any other 'atomic' field with custom limit
+    }
+  }  
+}
+```
+
+Default limits are equal to the one set in atomic iglucentral [schema](https://github.com/snowplow/iglu-central/blob/master/schemas/com.snowplowanalytics.snowplow/atomic/jsonschema/1-0-0), apart from:
+
+* `mkt_clickid` limit increased from 128 to 1000.
+* `page_url` limit increased from 4096 to 10000.
+* `page_referrer` limit increased from 4096 to 10000.
+
+Therefore, atomic fields with longer values could break downstream loading to data warehouse (Redshift/Snowflake/Databricks).
```

Comment:
In my understanding, Redshift would not break, because Redshift itself truncates the values. Also, we should detail what happens depending on the setting of `accept_invalid`